### PR TITLE
Prometheus: Change default httpMethod for new instances to POST

### LINKED ISSUE #30837: Make POST the default HTTP method for Prometheus datasource
**What would you like to be added**:
Make POST the default HTTP method for Prometheus datasource.

**Current behavior**:
The HTTP Method is not explicitly chosen in the datasource config UI, but defaults to GET:
image

**Why is this needed**:
POST allows for much larger query bodies that GET, which is needed when sending queries from graphs with a lot of targets (e.g., many hosts in a dashboard variable).

When queries fail due to the GET limitation on size of query body, it can be difficult for the Grafana user to troubleshoot, as the UI (and grafana.log) do not provide any info to indicate why the request has failed.

POST also makes the Query Inspector data easier to read, because the query is plaintext. (With GET, the query is URL encoded.)

In the past, it made sense to default to GET, because POST is only available in Prometheus >= v2.1.0 . But Prometheus 2.1.0 release is 3+ years old, so I think it makes sense to default to POST now.

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
This PR changes default http method for all of the newly added Prometheus data sources to 'POST'. (more in the comment directly in the code).

**Which issue(s) this PR fixes**:

Fixes: https://github.com/grafana/grafana/issues/30837

### CODE REVIEW

**Review by Elfo404 — APPROVED** [review]
Tested and seems to work as expected 🚀 

just left a super minor nit

**Review by davkal — APPROVED** [review]

File: public/app/plugins/datasource/prometheus/configuration/PromSettings.tsx [hunk 1]
```diff
@@ -64,7 +75,7 @@ export const PromSettings = (props: Props) => {
         <div className="gf-form">
           <InlineFormLabel
             width={13}
-            tooltip="Specify the HTTP Method to query Prometheus. (POST is only available in Prometheus >= v2.1.0)"
+            tooltip="You can use either POST or GET HTTP method to query your Prometheus data source. The POST method allows you to perform heavy requests, while the GET method will restrict you and return an error if the query is too large. POST is only available in Prometheus v2.1+)"
```
Comment (davkal) [inline public/app/plugins/datasource/prometheus/configuration/PromSettings.tsx:78]: I'd make this slightly more empowering:
```
POST is the recommended method as it allows bigger queries. Change this to GET if you have a Prometheus version older than 2.1 or if POST requests are restricted in your network.
```

Comment (ivanahuckova) [reply public/app/plugins/datasource/prometheus/configuration/PromSettings.tsx:78]:
```suggestion
            tooltip="You can use either POST or GET HTTP method to query your Prometheus data source. POST is the recommended method as it allows bigger queries. Change this to GET if you have a Prometheus version older than 2.1 or if POST requests are restricted in your network."
```

Comment (ivanahuckova) [reply public/app/plugins/datasource/prometheus/configuration/PromSettings.tsx]: Yes, I like your phrasing. Commited.

**Pull request merged**